src/Symfony/Component/Scheduler/RecurringMessage.php

```diff
@@ -42,30 +45,40 @@ private function __construct(
      */
     public static function every(string|int|\DateInterval $frequency, object $message, string|\DateTimeImmutable $from = new \DateTimeImmutable(), string|\DateTimeImmutable $until = new \DateTimeImmutable('3000-01-01')): self
     {
-        return new self(new PeriodicalTrigger($frequency, $from, $until), $message);
+        return self::trigger(new PeriodicalTrigger($frequency, $from, $until), $message);
     }
 
     public static function cron(string $expression, object $message, \DateTimeZone|string $timezone = null): self
     {
         if (!str_contains($expression, '#')) {
-            return new self(CronExpressionTrigger::fromSpec($expression, null, $timezone), $message);
+            return self::trigger(CronExpressionTrigger::fromSpec($expression, null, $timezone), $message);
         }
 
         if (!$message instanceof \Stringable) {
             throw new InvalidArgumentException('A message must be stringable to use "hashed" cron expressions.');
         }
 
-        return new self(CronExpressionTrigger::fromSpec($expression, (string) $message, $timezone), $message);
+        return self::trigger(CronExpressionTrigger::fromSpec($expression, (string) $message, $timezone), $message);
     }
 
     public static function trigger(TriggerInterface $trigger, object $message): self
     {
-        return new self($trigger, $message);
+        if ($message instanceof MessageProvider) {
+            return new self($trigger, $message);
+        }
+
+        try {
+            $description = $message instanceof \Stringable ? (string) $message : serialize($message);
```

Comment:
```suggestion
            $description = $message instanceof \Stringable ? (string) $message : '';
```
and remove the try/catch.

Reply:
I had to create a `Message` class in the tests since the Scheduler required their Id's to be unique. But I've applied the changes.

Reply:
Oh, my bad then, I forgot about this. So, let's revert to use `serialize` as before. Sorry for that.